<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Analysis | Differential Expression Analysis of BRCA samples</title>
  <meta name="description" content="4 Analysis | Differential Expression Analysis of BRCA samples" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Analysis | Differential Expression Analysis of BRCA samples" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="CrisSotomayor/rnaseq_project_LCG-UNAM_2022" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Analysis | Differential Expression Analysis of BRCA samples" />
  
  
  

<meta name="author" content="Cristina Sotomayor Vivas" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="differential-expression-analysis.html"/>

<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#requirements"><i class="fa fa-check"></i>Requirements</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="downloading-data.html"><a href="downloading-data.html"><i class="fa fa-check"></i><b>1</b> Downloading Data</a>
<ul>
<li class="chapter" data-level="1.1" data-path="downloading-data.html"><a href="downloading-data.html#downloading-through-recount-3"><i class="fa fa-check"></i><b>1.1</b> Downloading through recount 3</a></li>
<li class="chapter" data-level="1.2" data-path="downloading-data.html"><a href="downloading-data.html#obtaining-read-counts"><i class="fa fa-check"></i><b>1.2</b> Obtaining read counts</a></li>
<li class="chapter" data-level="1.3" data-path="downloading-data.html"><a href="downloading-data.html#exploring-data"><i class="fa fa-check"></i><b>1.3</b> Exploring data</a></li>
<li class="chapter" data-level="1.4" data-path="downloading-data.html"><a href="downloading-data.html#quality-control"><i class="fa fa-check"></i><b>1.4</b> Quality control</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="building-a-statistical-model.html"><a href="building-a-statistical-model.html"><i class="fa fa-check"></i><b>2</b> Building a statistical model</a>
<ul>
<li class="chapter" data-level="2.1" data-path="building-a-statistical-model.html"><a href="building-a-statistical-model.html#exploring-attributes"><i class="fa fa-check"></i><b>2.1</b> Exploring attributes</a></li>
<li class="chapter" data-level="2.2" data-path="building-a-statistical-model.html"><a href="building-a-statistical-model.html#normalizing-data"><i class="fa fa-check"></i><b>2.2</b> Normalizing data</a></li>
<li class="chapter" data-level="2.3" data-path="building-a-statistical-model.html"><a href="building-a-statistical-model.html#exploring-relationships-between-variables"><i class="fa fa-check"></i><b>2.3</b> Exploring relationships between variables</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html"><i class="fa fa-check"></i><b>3</b> Differential expression analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#normal-tissue"><i class="fa fa-check"></i><b>3.1</b> Normal tissue</a></li>
<li class="chapter" data-level="3.2" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#basal-subtype"><i class="fa fa-check"></i><b>3.2</b> Basal subtype</a></li>
<li class="chapter" data-level="3.3" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#her2-subtype"><i class="fa fa-check"></i><b>3.3</b> Her2 subtype</a></li>
<li class="chapter" data-level="3.4" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#luma-subtype"><i class="fa fa-check"></i><b>3.4</b> LumA subtype</a></li>
<li class="chapter" data-level="3.5" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#lumb-subtype"><i class="fa fa-check"></i><b>3.5</b> LumB subtype</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="analysis.html"><a href="analysis.html"><i class="fa fa-check"></i><b>4</b> Analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="analysis.html"><a href="analysis.html#differential-expression-by-chromosome"><i class="fa fa-check"></i><b>4.1</b> Differential Expression by Chromosome</a></li>
<li class="chapter" data-level="4.2" data-path="analysis.html"><a href="analysis.html#basal-tumors"><i class="fa fa-check"></i><b>4.2</b> Basal tumors</a></li>
<li class="chapter" data-level="4.3" data-path="analysis.html"><a href="analysis.html#her2-tumors"><i class="fa fa-check"></i><b>4.3</b> Her2 tumors</a></li>
<li class="chapter" data-level="4.4" data-path="analysis.html"><a href="analysis.html#luma-tumors"><i class="fa fa-check"></i><b>4.4</b> LumA tumors</a></li>
<li class="chapter" data-level="4.5" data-path="analysis.html"><a href="analysis.html#lumb-tumors"><i class="fa fa-check"></i><b>4.5</b> LumB tumors</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Differential Expression Analysis of BRCA samples</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="analysis" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Analysis</h1>
<p>Let’s now look at the genes we obtained and their functions.</p>
<ul>
<li>Basal: “CDCA8” “CAVIN2” “HJURP”</li>
<li>Her2: “RRM2” “HJURP” “KIF4A”</li>
<li>LumA: “LYVE1” “VEGFD” “DMD”<br />
</li>
<li>LumB: “PAMR1” “CAVIN2” “DMD”</li>
</ul>
<p>Looking at what each of the genes does:
- <a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=CDCA8">CDCA8</a>: Cell Division Cycle Associated 8. It is a component of the chromosomal passenger complex (CPC), a complex that acts as a key regulator of mitosis.
- <a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=CAVIN2&amp;keywords=cavin2">CAVIN2</a>: Caveolae Associated Protein 2, plays an important role in caveolar biogenesis and morphology.
- <a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=HJURP">HJURP</a>: Holliday Junction Recognition Protein, participates in DNA damage and mitotic cell cycle pathways. Genecards mentions it is associated with Luminal A subtype.
- <a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=RRM2">RRM2</a>: Ribonucleotide Reductase Regulatory Subunit M2, catalyzes the biosynthesis of deoxyribonucleotides from the corresponding ribonucleotides, necessary for DNA synthesis.
- <a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=KIF4A">KIF4A</a>: Kinesin Family Member 4A, ATP dependent microtubule-based motor protein that is involved in the intracellular transport of membranous organelles.
- <a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=LYVE1">LYVE1</a>: Lymphatic Vessel Endothelial Hyaluronan Receptor 1, plays a role in autocrine regulation of cell growth.
- <a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=VEGFD">VEGFD</a>: Vascular Endothelial Growth Factor D, growth factor active in angiogenesis, lymphangiogenesis and endothelial cell growth, stimulating their proliferation and migration and also has effects on the permeability of blood vessels.
- <a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=DMD">DMD</a>: Dystrophin, forms a component of the dystrophin-glycoprotein complex (DGC), which bridges the inner cytoskeleton and the extracellular matrix.
- <a href="https://www.genecards.org/cgi-bin/carddisp.pl?gene=PAMR1">PAMR1</a>: Peptidase Domain Containing Associated With Muscle Regeneration 1, may play a role in regeneration of skeletal muscle.</p>
<p>We can see that most of these genes are related to cell growth and mitotic cell division, two pathways that are commonly misregulated in cancer. One, HJURP, is associated with Luminal A breast cancer, but in our analysis turned out to be more significant for Basal and Her2 subtypes. Interestingly, of the 9 different genes considered here, only one, “RRM2” is among the 50 genes in the PAM50 list defining the subtypes. However, we only considered the three with lowest values for each subtype, expanding this analysis to more genes may result in more genes in the list arising.</p>
<p>We can visualize the difference in expression of these genes through a heatmap, using the <code>pheatmap</code> library.</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="analysis.html#cb97-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;pheatmap&quot;</span>)</span>
<span id="cb97-2"><a href="analysis.html#cb97-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-3"><a href="analysis.html#cb97-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Col data</span></span>
<span id="cb97-4"><a href="analysis.html#cb97-4" aria-hidden="true" tabindex="-1"></a>df_col<span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(rse_BRCA<span class="sc">$</span>subtype)</span>
<span id="cb97-5"><a href="analysis.html#cb97-5" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(df_col) <span class="ot">&lt;-</span> <span class="st">&quot;Subtype&quot;</span></span>
<span id="cb97-6"><a href="analysis.html#cb97-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(df_col) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(rse_BRCA)</span>
<span id="cb97-7"><a href="analysis.html#cb97-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-8"><a href="analysis.html#cb97-8" aria-hidden="true" tabindex="-1"></a><span class="co"># PAM50 genes</span></span>
<span id="cb97-9"><a href="analysis.html#cb97-9" aria-hidden="true" tabindex="-1"></a>PAM50 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;UBE2T&quot;</span>,<span class="st">&quot;BIRC5&quot;</span>,<span class="st">&quot;NUF2&quot;</span>,<span class="st">&quot;CDC6&quot;</span>,<span class="st">&quot;CCNB1&quot;</span>,<span class="st">&quot;TYMS&quot;</span>,<span class="st">&quot;MYBL2&quot;</span>,<span class="st">&quot;CEP55&quot;</span>,<span class="st">&quot;MELK&quot;</span>,<span class="st">&quot;NDC80&quot;</span>,<span class="st">&quot;RRM2&quot;</span>,<span class="st">&quot;UBE2C&quot;</span>,<span class="st">&quot;CENPF&quot;</span>,<span class="st">&quot;PTTG1&quot;</span>,<span class="st">&quot;EXO1&quot;</span>,<span class="st">&quot;ORC6&quot;</span>,<span class="st">&quot;ANLN&quot;</span>,<span class="st">&quot;CCNE1&quot;</span>,<span class="st">&quot;CDC20&quot;</span>,<span class="st">&quot;MKI67&quot;</span>,<span class="st">&quot;KIF2C&quot;</span>,<span class="st">&quot;ACTR3B&quot;</span> ,<span class="st">&quot;MYC&quot;</span>,<span class="st">&quot;EGFR&quot;</span>,<span class="st">&quot;KRT5&quot;</span>,<span class="st">&quot;PHGDH&quot;</span>,<span class="st">&quot;CDH3&quot;</span>,<span class="st">&quot;MIA&quot;</span>,<span class="st">&quot;KRT17&quot;</span>,<span class="st">&quot;FOXC1&quot;</span>,<span class="st">&quot;SFRP1&quot;</span>,<span class="st">&quot;KRT14&quot;</span>,<span class="st">&quot;ESR1&quot;</span>,<span class="st">&quot;SLC39A6&quot;</span>,<span class="st">&quot;BAG1&quot;</span>,<span class="st">&quot;MAPT&quot;</span>,<span class="st">&quot;PGR&quot;</span>,<span class="st">&quot;CXXC5&quot;</span>,<span class="st">&quot;MLPH&quot;</span>,<span class="st">&quot;BCL2&quot;</span>,<span class="st">&quot;MDM2&quot;</span>,<span class="st">&quot;NAT1&quot;</span>,<span class="st">&quot;FOXA1&quot;</span>,<span class="st">&quot;BLVRA&quot;</span>,<span class="st">&quot;MMP11&quot;</span>,<span class="st">&quot;GPR160&quot;</span>,<span class="st">&quot;FGFR4&quot;</span>,<span class="st">&quot;GRB7&quot;</span>,<span class="st">&quot;TMEM45B&quot;</span>,<span class="st">&quot;ERBB2&quot;</span>)</span>
<span id="cb97-10"><a href="analysis.html#cb97-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-11"><a href="analysis.html#cb97-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Matching genes to ids</span></span>
<span id="cb97-12"><a href="analysis.html#cb97-12" aria-hidden="true" tabindex="-1"></a>PAM50_ids <span class="ot">&lt;-</span> <span class="fu">match</span>(PAM50, <span class="fu">rowData</span>(rse_BRCA)<span class="sc">$</span>gene_name)</span>
<span id="cb97-13"><a href="analysis.html#cb97-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-14"><a href="analysis.html#cb97-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Filtering table </span></span>
<span id="cb97-15"><a href="analysis.html#cb97-15" aria-hidden="true" tabindex="-1"></a>PAM50_exp <span class="ot">&lt;-</span> vGene<span class="sc">$</span>E[PAM50_ids, ]</span>
<span id="cb97-16"><a href="analysis.html#cb97-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb97-17"><a href="analysis.html#cb97-17" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(</span>
<span id="cb97-18"><a href="analysis.html#cb97-18" aria-hidden="true" tabindex="-1"></a>  PAM50_exp, </span>
<span id="cb97-19"><a href="analysis.html#cb97-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">TRUE</span>,</span>
<span id="cb97-20"><a href="analysis.html#cb97-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>, </span>
<span id="cb97-21"><a href="analysis.html#cb97-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_rownames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb97-22"><a href="analysis.html#cb97-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_colnames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb97-23"><a href="analysis.html#cb97-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_col =</span> df_col</span>
<span id="cb97-24"><a href="analysis.html#cb97-24" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-42-1.png" width="672" /></p>
<p>We can see that the subtypes do not all cluster together, even with the genes that differentiate between them. The Basal subtype seems to be the most different from the rest, and it is also the most aggressive of all subtypes, with the worst clinical outcome.</p>
<div id="differential-expression-by-chromosome" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Differential Expression by Chromosome</h2>
<p>Lastly, we will look at which chromosomes contain the most differentially expressed genes by subtype, focusing on the 1K protein-coding genes with smallest adjusted p-values.</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="analysis.html#cb98-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to extract expression of protein coding genes with lowest p vals</span></span>
<span id="cb98-2"><a href="analysis.html#cb98-2" aria-hidden="true" tabindex="-1"></a>diff_exp_genes <span class="ot">&lt;-</span> <span class="cf">function</span>(de_results, <span class="at">n=</span><span class="dv">1000</span>) {</span>
<span id="cb98-3"><a href="analysis.html#cb98-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb98-4"><a href="analysis.html#cb98-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filtering protein coding</span></span>
<span id="cb98-5"><a href="analysis.html#cb98-5" aria-hidden="true" tabindex="-1"></a>  protein_coding <span class="ot">&lt;-</span> de_results[de_results<span class="sc">$</span>gene_type <span class="sc">==</span> <span class="st">&#39;protein_coding&#39;</span>, ]</span>
<span id="cb98-6"><a href="analysis.html#cb98-6" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb98-7"><a href="analysis.html#cb98-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Selecting top n</span></span>
<span id="cb98-8"><a href="analysis.html#cb98-8" aria-hidden="true" tabindex="-1"></a>  smallest_p_vals <span class="ot">&lt;-</span> protein_coding<span class="sc">$</span>gene_id[<span class="fu">rank</span>(protein_coding<span class="sc">$</span>adj.P.Val) <span class="sc">&lt;=</span> n]</span>
<span id="cb98-9"><a href="analysis.html#cb98-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb98-10"><a href="analysis.html#cb98-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Getting these genes for heatmap </span></span>
<span id="cb98-11"><a href="analysis.html#cb98-11" aria-hidden="true" tabindex="-1"></a>  exprs_heatmap <span class="ot">&lt;-</span> vGene<span class="sc">$</span>E[<span class="fu">match</span>(smallest_p_vals, <span class="fu">rownames</span>(vGene)), ]</span>
<span id="cb98-12"><a href="analysis.html#cb98-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb98-13"><a href="analysis.html#cb98-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(exprs_heatmap)</span>
<span id="cb98-14"><a href="analysis.html#cb98-14" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>To visualize how many differentially expressed genes there are in each chromosome, we can use the chromosome data from <code>seqnames</code> in <code>rowRanges(rse_BRCA)</code>.</p>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="analysis.html#cb99-1" aria-hidden="true" tabindex="-1"></a>chr_df <span class="ot">&lt;-</span> <span class="cf">function</span>(exprs) {</span>
<span id="cb99-2"><a href="analysis.html#cb99-2" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb99-3"><a href="analysis.html#cb99-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get genes from row names </span></span>
<span id="cb99-4"><a href="analysis.html#cb99-4" aria-hidden="true" tabindex="-1"></a>  genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(exprs)</span>
<span id="cb99-5"><a href="analysis.html#cb99-5" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb99-6"><a href="analysis.html#cb99-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Access chr name from rowRanges</span></span>
<span id="cb99-7"><a href="analysis.html#cb99-7" aria-hidden="true" tabindex="-1"></a>  chrs <span class="ot">&lt;-</span> <span class="fu">seqnames</span>(<span class="fu">rowRanges</span>(rse_BRCA)[genes])</span>
<span id="cb99-8"><a href="analysis.html#cb99-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb99-9"><a href="analysis.html#cb99-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Drop unused levels, count occurrences </span></span>
<span id="cb99-10"><a href="analysis.html#cb99-10" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">table</span>(<span class="fu">droplevels</span>(chrs))</span>
<span id="cb99-11"><a href="analysis.html#cb99-11" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(df)</span>
<span id="cb99-12"><a href="analysis.html#cb99-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Chromosome&quot;</span>, <span class="st">&quot;Frequency&quot;</span>)</span>
<span id="cb99-13"><a href="analysis.html#cb99-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb99-14"><a href="analysis.html#cb99-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(df)</span>
<span id="cb99-15"><a href="analysis.html#cb99-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb99-16"><a href="analysis.html#cb99-16" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="basal-tumors" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Basal tumors</h2>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="analysis.html#cb100-1" aria-hidden="true" tabindex="-1"></a>diff_basal <span class="ot">&lt;-</span> <span class="fu">diff_exp_genes</span>(de_results_basal)</span>
<span id="cb100-2"><a href="analysis.html#cb100-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-3"><a href="analysis.html#cb100-3" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(</span>
<span id="cb100-4"><a href="analysis.html#cb100-4" aria-hidden="true" tabindex="-1"></a>    diff_basal, </span>
<span id="cb100-5"><a href="analysis.html#cb100-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_cols =</span> <span class="cn">TRUE</span>,</span>
<span id="cb100-6"><a href="analysis.html#cb100-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>, </span>
<span id="cb100-7"><a href="analysis.html#cb100-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">show_rownames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb100-8"><a href="analysis.html#cb100-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">show_colnames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb100-9"><a href="analysis.html#cb100-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">annotation_col =</span> df_col</span>
<span id="cb100-10"><a href="analysis.html#cb100-10" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-45-1.png" width="672" /></p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="analysis.html#cb101-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">chr_df</span>(diff_basal)</span>
<span id="cb101-2"><a href="analysis.html#cb101-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb101-3"><a href="analysis.html#cb101-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x=</span>Chromosome, <span class="at">y=</span>Frequency)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="at">fill=</span><span class="st">&quot;palegreen4&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-46-1.png" width="672" /></p>
</div>
<div id="her2-tumors" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Her2 tumors</h2>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="analysis.html#cb102-1" aria-hidden="true" tabindex="-1"></a>diff_her2 <span class="ot">&lt;-</span> <span class="fu">diff_exp_genes</span>(de_results_her2)</span>
<span id="cb102-2"><a href="analysis.html#cb102-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb102-3"><a href="analysis.html#cb102-3" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(</span>
<span id="cb102-4"><a href="analysis.html#cb102-4" aria-hidden="true" tabindex="-1"></a>    diff_her2, </span>
<span id="cb102-5"><a href="analysis.html#cb102-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_cols =</span> <span class="cn">TRUE</span>,</span>
<span id="cb102-6"><a href="analysis.html#cb102-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>, </span>
<span id="cb102-7"><a href="analysis.html#cb102-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">show_rownames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb102-8"><a href="analysis.html#cb102-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">show_colnames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb102-9"><a href="analysis.html#cb102-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">annotation_col =</span> df_col</span>
<span id="cb102-10"><a href="analysis.html#cb102-10" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-47-1.png" width="672" /></p>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="analysis.html#cb103-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">chr_df</span>(diff_her2)</span>
<span id="cb103-2"><a href="analysis.html#cb103-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb103-3"><a href="analysis.html#cb103-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x=</span>Chromosome, <span class="at">y=</span>Frequency)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="at">fill=</span><span class="st">&quot;palegreen4&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-48-1.png" width="672" /></p>
</div>
<div id="luma-tumors" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> LumA tumors</h2>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="analysis.html#cb104-1" aria-hidden="true" tabindex="-1"></a>diff_luma <span class="ot">&lt;-</span> <span class="fu">diff_exp_genes</span>(de_results_luma)</span>
<span id="cb104-2"><a href="analysis.html#cb104-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb104-3"><a href="analysis.html#cb104-3" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(</span>
<span id="cb104-4"><a href="analysis.html#cb104-4" aria-hidden="true" tabindex="-1"></a>    diff_luma, </span>
<span id="cb104-5"><a href="analysis.html#cb104-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_cols =</span> <span class="cn">TRUE</span>,</span>
<span id="cb104-6"><a href="analysis.html#cb104-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>, </span>
<span id="cb104-7"><a href="analysis.html#cb104-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">show_rownames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb104-8"><a href="analysis.html#cb104-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">show_colnames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb104-9"><a href="analysis.html#cb104-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">annotation_col =</span> df_col</span>
<span id="cb104-10"><a href="analysis.html#cb104-10" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-49-1.png" width="672" /></p>
<div class="sourceCode" id="cb105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb105-1"><a href="analysis.html#cb105-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">chr_df</span>(diff_luma)</span>
<span id="cb105-2"><a href="analysis.html#cb105-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb105-3"><a href="analysis.html#cb105-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x=</span>Chromosome, <span class="at">y=</span>Frequency)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="at">fill=</span><span class="st">&quot;palegreen4&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-50-1.png" width="672" /></p>
</div>
<div id="lumb-tumors" class="section level2" number="4.5">
<h2><span class="header-section-number">4.5</span> LumB tumors</h2>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="analysis.html#cb106-1" aria-hidden="true" tabindex="-1"></a>diff_lumb <span class="ot">&lt;-</span> <span class="fu">diff_exp_genes</span>(de_results_lumb)</span>
<span id="cb106-2"><a href="analysis.html#cb106-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-3"><a href="analysis.html#cb106-3" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(</span>
<span id="cb106-4"><a href="analysis.html#cb106-4" aria-hidden="true" tabindex="-1"></a>    diff_lumb, </span>
<span id="cb106-5"><a href="analysis.html#cb106-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_cols =</span> <span class="cn">TRUE</span>,</span>
<span id="cb106-6"><a href="analysis.html#cb106-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>, </span>
<span id="cb106-7"><a href="analysis.html#cb106-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">show_rownames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb106-8"><a href="analysis.html#cb106-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">show_colnames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb106-9"><a href="analysis.html#cb106-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">annotation_col =</span> df_col</span>
<span id="cb106-10"><a href="analysis.html#cb106-10" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-51-1.png" width="672" /></p>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="analysis.html#cb107-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">chr_df</span>(diff_lumb)</span>
<span id="cb107-2"><a href="analysis.html#cb107-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb107-3"><a href="analysis.html#cb107-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x=</span>Chromosome, <span class="at">y=</span>Frequency)) <span class="sc">+</span> <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>, <span class="at">fill=</span><span class="st">&quot;palegreen4&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-52-1.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="differential-expression-analysis.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
