<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>1 Downloading Data | Differential Expression Analysis of BRCA samples</title>
  <meta name="description" content="1 Downloading Data | Differential Expression Analysis of BRCA samples" />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="1 Downloading Data | Differential Expression Analysis of BRCA samples" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="CrisSotomayor/rnaseq_project_LCG-UNAM_2022" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="1 Downloading Data | Differential Expression Analysis of BRCA samples" />
  
  
  

<meta name="author" content="Cristina Sotomayor Vivas" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="building-a-statistical-model.html"/>
<script src="libs/header-attrs-2.6/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#requirements"><i class="fa fa-check"></i>Requirements</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="downloading-data.html"><a href="downloading-data.html"><i class="fa fa-check"></i><b>1</b> Downloading Data</a>
<ul>
<li class="chapter" data-level="1.1" data-path="downloading-data.html"><a href="downloading-data.html#downloading-through-recount-3"><i class="fa fa-check"></i><b>1.1</b> Downloading through recount 3</a></li>
<li class="chapter" data-level="1.2" data-path="downloading-data.html"><a href="downloading-data.html#obtaining-read-counts"><i class="fa fa-check"></i><b>1.2</b> Obtaining read counts</a></li>
<li class="chapter" data-level="1.3" data-path="downloading-data.html"><a href="downloading-data.html#exploring-data"><i class="fa fa-check"></i><b>1.3</b> Exploring data</a></li>
<li class="chapter" data-level="1.4" data-path="downloading-data.html"><a href="downloading-data.html#quality-control"><i class="fa fa-check"></i><b>1.4</b> Quality control</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="building-a-statistical-model.html"><a href="building-a-statistical-model.html"><i class="fa fa-check"></i><b>2</b> Building a statistical model</a>
<ul>
<li class="chapter" data-level="2.1" data-path="building-a-statistical-model.html"><a href="building-a-statistical-model.html#exploring-attributes"><i class="fa fa-check"></i><b>2.1</b> Exploring attributes</a></li>
<li class="chapter" data-level="2.2" data-path="building-a-statistical-model.html"><a href="building-a-statistical-model.html#normalizing-data"><i class="fa fa-check"></i><b>2.2</b> Normalizing data</a></li>
<li class="chapter" data-level="2.3" data-path="building-a-statistical-model.html"><a href="building-a-statistical-model.html#exploring-relationships-between-variables"><i class="fa fa-check"></i><b>2.3</b> Exploring relationships between variables</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html"><i class="fa fa-check"></i><b>3</b> Differential expression analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#normal-tissue"><i class="fa fa-check"></i><b>3.1</b> Normal tissue</a></li>
<li class="chapter" data-level="3.2" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#basal-subtype"><i class="fa fa-check"></i><b>3.2</b> Basal subtype</a></li>
<li class="chapter" data-level="3.3" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#her2-subtype"><i class="fa fa-check"></i><b>3.3</b> Her2 subtype</a></li>
<li class="chapter" data-level="3.4" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#luma-subtype"><i class="fa fa-check"></i><b>3.4</b> LumA subtype</a></li>
<li class="chapter" data-level="3.5" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#lumb-subtype"><i class="fa fa-check"></i><b>3.5</b> LumB subtype</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="analysis.html"><a href="analysis.html"><i class="fa fa-check"></i><b>4</b> Analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="analysis.html"><a href="analysis.html#differential-expression-by-chromosome"><i class="fa fa-check"></i><b>4.1</b> Differential Expression by Chromosome</a></li>
<li class="chapter" data-level="4.2" data-path="analysis.html"><a href="analysis.html#basal-tumors"><i class="fa fa-check"></i><b>4.2</b> Basal tumors</a></li>
<li class="chapter" data-level="4.3" data-path="analysis.html"><a href="analysis.html#her2-tumors"><i class="fa fa-check"></i><b>4.3</b> Her2 tumors</a></li>
<li class="chapter" data-level="4.4" data-path="analysis.html"><a href="analysis.html#luma-tumors"><i class="fa fa-check"></i><b>4.4</b> LumA tumors</a></li>
<li class="chapter" data-level="4.5" data-path="analysis.html"><a href="analysis.html#lumb-tumors"><i class="fa fa-check"></i><b>4.5</b> LumB tumors</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Differential Expression Analysis of BRCA samples</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="downloading-data" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Downloading Data</h1>
<p>The first step in the project is obtaining the data that we will analyze, as well as processing it so it is in the format we need for further study.</p>
<div id="downloading-through-recount-3" class="section level2" number="1.1">
<h2><span class="header-section-number">1.1</span> Downloading through recount 3</h2>
<p><a href="http://rna.recount.bio"><code>recount3</code></a> is a project that provides access to over 750,000 publicly available human and mouse RNA-seq samples, all uniformly processed. The original documentation can be found <a href="http://rna.recount.bio/docs/index.html">here</a>.</p>
<p>For this project, we will be working with BRCA data from The Cancer Genome Atlas Program (<a href="https://www.cancer.gov/about-nci/organization/ccg/research/structural-genomics/tcga">TCGA</a>). We will begin by loading the <code>recount3</code> package and its dependencies.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="downloading-data.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Load recount3 R package</span></span>
<span id="cb6-2"><a href="downloading-data.html#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;recount3&quot;</span>)</span></code></pre></div>
<p>The data available in the <code>recount3</code> project can be explored <a href="https://jhubiostatistics.shinyapps.io/recount3-study-explorer/">here</a>. To download the data we are interested in, we will use the following code.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="downloading-data.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create RangedSummarizedExperiment (RSE) object</span></span>
<span id="cb7-2"><a href="downloading-data.html#cb7-2" aria-hidden="true" tabindex="-1"></a>rse_BRCA <span class="ot">&lt;-</span> recount3<span class="sc">::</span><span class="fu">create_rse_manual</span>(</span>
<span id="cb7-3"><a href="downloading-data.html#cb7-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">project =</span> <span class="st">&quot;BRCA&quot;</span>,</span>
<span id="cb7-4"><a href="downloading-data.html#cb7-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">project_home =</span> <span class="st">&quot;data_sources/tcga&quot;</span>,</span>
<span id="cb7-5"><a href="downloading-data.html#cb7-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">organism =</span> <span class="st">&quot;human&quot;</span>,</span>
<span id="cb7-6"><a href="downloading-data.html#cb7-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">annotation =</span> <span class="st">&quot;gencode_v29&quot;</span>,</span>
<span id="cb7-7"><a href="downloading-data.html#cb7-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">type =</span> <span class="st">&quot;gene&quot;</span></span>
<span id="cb7-8"><a href="downloading-data.html#cb7-8" aria-hidden="true" tabindex="-1"></a>          )</span>
<span id="cb7-9"><a href="downloading-data.html#cb7-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-10"><a href="downloading-data.html#cb7-10" aria-hidden="true" tabindex="-1"></a>rse_BRCA</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 64837 1256 
## metadata(8): time_created recount3_version ... annotation recount3_url
## assays(1): raw_counts
## rownames(64837): ENSG00000278704.1 ENSG00000277400.1 ... ENSG00000182484.15_PAR_Y ENSG00000227159.8_PAR_Y
## rowData names(10): source type ... havana_gene tag
## colnames(1256): 0b1cf6de-f98f-4020-baa7-c4fc518e5f00 20c90e97-e900-488e-af6a-6648d9a8bb01 ...
##   a340e38d-af36-4b13-add3-50be6819d4fe a79dd5b2-f97a-4b24-b762-fea1cd5238fe
## colData names(937): rail_id external_id ... recount_seq_qc.errq BigWigURL</code></pre>
<p>This data set includes 63856 genes, and 1256 samples.</p>
</div>
<div id="obtaining-read-counts" class="section level2" number="1.2">
<h2><span class="header-section-number">1.2</span> Obtaining read counts</h2>
<p>The <code>RSE</code> object contains the raw base-pair counts, to compute the read counts, we use the following code.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="downloading-data.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Saving read counts as an assay in the rse object</span></span>
<span id="cb9-2"><a href="downloading-data.html#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(rse_BRCA, <span class="st">&quot;counts&quot;</span>) <span class="ot">&lt;-</span> <span class="fu">compute_read_counts</span>(rse_BRCA)</span></code></pre></div>
</div>
<div id="exploring-data" class="section level2" number="1.3">
<h2><span class="header-section-number">1.3</span> Exploring data</h2>
<p>Our data is stored as a <code>RangedSummarizedExperiment</code> object, which includes data for both the samples (columns), and the genes (rows), as well as metadata and the assays (counts, in this case). We will first perform quality control, and then select some sample attributes for our model.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="downloading-data.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Number of samples and number of sample attributes </span></span>
<span id="cb10-2"><a href="downloading-data.html#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(<span class="fu">colData</span>(rse_BRCA))</span></code></pre></div>
<pre><code>## [1] 1256  937</code></pre>
<p>For our 1256 samples, there are 937 columns, corresponding to 937 possible attributes to include in our model. To reduce the options, we will first filter by removing those attributes with only NA values.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="downloading-data.html#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only non-empty columns</span></span>
<span id="cb12-2"><a href="downloading-data.html#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(rse_BRCA) <span class="ot">&lt;-</span> <span class="fu">colData</span>(rse_BRCA)[, <span class="fu">colSums</span>(<span class="fu">is.na</span>(<span class="fu">colData</span>(rse_BRCA))) <span class="sc">!=</span> <span class="fu">nrow</span>(<span class="fu">colData</span>(rse_BRCA))]</span>
<span id="cb12-3"><a href="downloading-data.html#cb12-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-4"><a href="downloading-data.html#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(<span class="fu">colData</span>(rse_BRCA))</span></code></pre></div>
<pre><code>## [1] 1256  388</code></pre>
<p>This leaves us with 388 columns. These attributes have terms separated by periods, we can explore the source of each attribute by selecting the first keyword of each one to see what kind of information we have.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="downloading-data.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Split each string by &#39;.&#39;</span></span>
<span id="cb14-2"><a href="downloading-data.html#cb14-2" aria-hidden="true" tabindex="-1"></a>split_col_names <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(<span class="fu">colnames</span>(<span class="fu">colData</span>(rse_BRCA)), <span class="st">&quot;.&quot;</span>, <span class="at">fixed=</span><span class="cn">TRUE</span>)</span>
<span id="cb14-3"><a href="downloading-data.html#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="downloading-data.html#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep only first element </span></span>
<span id="cb14-5"><a href="downloading-data.html#cb14-5" aria-hidden="true" tabindex="-1"></a>split_col_names <span class="ot">&lt;-</span> <span class="fu">sapply</span>(split_col_names, getElement, <span class="dv">1</span>)</span>
<span id="cb14-6"><a href="downloading-data.html#cb14-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-7"><a href="downloading-data.html#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Counting the results</span></span>
<span id="cb14-8"><a href="downloading-data.html#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(split_col_names)</span></code></pre></div>
<pre><code>## split_col_names
##       BigWigURL     external_id         rail_id recount_project      recount_qc  recount_seq_qc           study 
##               1               1               1               5              79              12               1 
##            tcga 
##             288</code></pre>
<p>Our main sources of information are from <code>recount</code> and from <code>tcga</code>.</p>
</div>
<div id="quality-control" class="section level2" number="1.4">
<h2><span class="header-section-number">1.4</span> Quality control</h2>
<p>Letâ€™s look at both <code>recount_qc</code> and <code>recount_seq_qc</code> for quality control. The meaning of the different fields can be found in the <a href="http://rna.recount.bio/docs/quality-check-fields.html">documentation</a>.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="downloading-data.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(<span class="fu">colData</span>(rse_BRCA))[<span class="fu">grepl</span>(<span class="st">&#39;^recount_seq_qc&#39;</span>, <span class="fu">colnames</span>(<span class="fu">colData</span>(rse_BRCA)))]</span></code></pre></div>
<pre><code>##  [1] &quot;recount_seq_qc.min_len&quot;                  &quot;recount_seq_qc.max_len&quot;                 
##  [3] &quot;recount_seq_qc.avg_len&quot;                  &quot;recount_seq_qc.#distinct_quality_values&quot;
##  [5] &quot;recount_seq_qc.#bases&quot;                   &quot;recount_seq_qc.%a&quot;                      
##  [7] &quot;recount_seq_qc.%c&quot;                       &quot;recount_seq_qc.%g&quot;                      
##  [9] &quot;recount_seq_qc.%t&quot;                       &quot;recount_seq_qc.%n&quot;                      
## [11] &quot;recount_seq_qc.avgq&quot;                     &quot;recount_seq_qc.errq&quot;</code></pre>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="downloading-data.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">colnames</span>(<span class="fu">colData</span>(rse_BRCA))[<span class="fu">grepl</span>(<span class="st">&#39;^recount_qc&#39;</span>, <span class="fu">colnames</span>(<span class="fu">colData</span>(rse_BRCA)))], <span class="dv">10</span>)</span></code></pre></div>
<pre><code>##  [1] &quot;recount_qc.aligned_reads%.chrm&quot;                 &quot;recount_qc.aligned_reads%.chrx&quot;                
##  [3] &quot;recount_qc.aligned_reads%.chry&quot;                 &quot;recount_qc.bc_auc.all_reads_all_bases&quot;         
##  [5] &quot;recount_qc.bc_auc.all_reads_annotated_bases&quot;    &quot;recount_qc.bc_auc.unique_reads_all_bases&quot;      
##  [7] &quot;recount_qc.bc_auc.unique_reads_annotated_bases&quot; &quot;recount_qc.bc_auc.all_%&quot;                       
##  [9] &quot;recount_qc.bc_auc.unique_%&quot;                     &quot;recount_qc.bc_frag.count&quot;</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="downloading-data.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># The field `recount_seq_qc.avgq` corresponds to the weighted average over Phred quality scores</span></span>
<span id="cb20-2"><a href="downloading-data.html#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(rse_BRCA<span class="sc">$</span>recount_seq_qc.avgq)</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   21.10   36.90   37.40   37.31   37.90   39.00</code></pre>
<p>All Phred scores are above 20, meaning all samples are estimated to be at least 99% accurate.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="downloading-data.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Exploring unmapped reads </span></span>
<span id="cb22-2"><a href="downloading-data.html#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(rse_BRCA<span class="sc">$</span><span class="st">&#39;recount_qc.star.%_of_reads_unmapped:_other&#39;</span>)</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
## 0.01000 0.03000 0.04000 0.05249 0.05000 1.31000</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="downloading-data.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(rse_BRCA<span class="sc">$</span><span class="st">&#39;recount_qc.star.%_of_reads_unmapped:_too_many_mismatches&#39;</span>)</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##       0       0       0       0       0       0</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="downloading-data.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(rse_BRCA<span class="sc">$</span><span class="st">&#39;recount_qc.star.%_of_reads_unmapped:_too_short&#39;</span>)</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##   2.030   4.990   5.650   6.319   6.522  56.120</code></pre>
<p>There are few unmapped reads, except for some that are too short.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="downloading-data.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Exploring mismatch rate</span></span>
<span id="cb28-2"><a href="downloading-data.html#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(rse_BRCA<span class="sc">$</span><span class="st">&#39;recount_qc.star.mismatch_rate_per_base,_%&#39;</span>)</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##  0.1400  0.2500  0.2900  0.3026  0.3400  0.7100</code></pre>
<p>Very small percentage of mismatches per base. Overall, it seems that the samples are good quality, and we will proceed with the selection of attributes for our model.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="building-a-statistical-model.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
