<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Differential Expression Analysis of BRCA samples</title>
  <meta name="description" content="Differential Expression Analysis of BRCA samples" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Differential Expression Analysis of BRCA samples" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="CrisSotomayor/rnaseq_project_LCG-UNAM_2022" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Differential Expression Analysis of BRCA samples" />
  
  
  

<meta name="author" content="Cristina Sotomayor Vivas" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="downloading-data.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#requirements"><i class="fa fa-check"></i>Requirements</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="downloading-data.html"><a href="downloading-data.html"><i class="fa fa-check"></i><b>1</b> Downloading Data</a>
<ul>
<li class="chapter" data-level="1.1" data-path="downloading-data.html"><a href="downloading-data.html#downloading-through-recount-3"><i class="fa fa-check"></i><b>1.1</b> Downloading through recount 3</a></li>
<li class="chapter" data-level="1.2" data-path="downloading-data.html"><a href="downloading-data.html#obtaining-read-counts"><i class="fa fa-check"></i><b>1.2</b> Obtaining read counts</a></li>
<li class="chapter" data-level="1.3" data-path="downloading-data.html"><a href="downloading-data.html#exploring-data"><i class="fa fa-check"></i><b>1.3</b> Exploring data</a></li>
<li class="chapter" data-level="1.4" data-path="downloading-data.html"><a href="downloading-data.html#quality-control"><i class="fa fa-check"></i><b>1.4</b> Quality control</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="building-a-statistical-model.html"><a href="building-a-statistical-model.html"><i class="fa fa-check"></i><b>2</b> Building a statistical model</a>
<ul>
<li class="chapter" data-level="2.1" data-path="building-a-statistical-model.html"><a href="building-a-statistical-model.html#exploring-attributes"><i class="fa fa-check"></i><b>2.1</b> Exploring attributes</a></li>
<li class="chapter" data-level="2.2" data-path="building-a-statistical-model.html"><a href="building-a-statistical-model.html#normalizing-data"><i class="fa fa-check"></i><b>2.2</b> Normalizing data</a></li>
<li class="chapter" data-level="2.3" data-path="building-a-statistical-model.html"><a href="building-a-statistical-model.html#exploring-relationships-between-variables"><i class="fa fa-check"></i><b>2.3</b> Exploring relationships between variables</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html"><i class="fa fa-check"></i><b>3</b> Differential expression analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#normal-tissue"><i class="fa fa-check"></i><b>3.1</b> Normal tissue</a></li>
<li class="chapter" data-level="3.2" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#basal-subtype"><i class="fa fa-check"></i><b>3.2</b> Basal subtype</a></li>
<li class="chapter" data-level="3.3" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#her2-subtype"><i class="fa fa-check"></i><b>3.3</b> Her2 subtype</a></li>
<li class="chapter" data-level="3.4" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#luma-subtype"><i class="fa fa-check"></i><b>3.4</b> LumA subtype</a></li>
<li class="chapter" data-level="3.5" data-path="differential-expression-analysis.html"><a href="differential-expression-analysis.html#lumb-subtype"><i class="fa fa-check"></i><b>3.5</b> LumB subtype</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="analysis.html"><a href="analysis.html"><i class="fa fa-check"></i><b>4</b> Analysis</a>
<ul>
<li class="chapter" data-level="4.1" data-path="analysis.html"><a href="analysis.html#differential-expression-by-chromosome"><i class="fa fa-check"></i><b>4.1</b> Differential Expression by Chromosome</a></li>
<li class="chapter" data-level="4.2" data-path="analysis.html"><a href="analysis.html#basal-tumors"><i class="fa fa-check"></i><b>4.2</b> Basal tumors</a></li>
<li class="chapter" data-level="4.3" data-path="analysis.html"><a href="analysis.html#her2-tumors"><i class="fa fa-check"></i><b>4.3</b> Her2 tumors</a></li>
<li class="chapter" data-level="4.4" data-path="analysis.html"><a href="analysis.html#luma-tumors"><i class="fa fa-check"></i><b>4.4</b> LumA tumors</a></li>
<li class="chapter" data-level="4.5" data-path="analysis.html"><a href="analysis.html#lumb-tumors"><i class="fa fa-check"></i><b>4.5</b> LumB tumors</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Differential Expression Analysis of BRCA samples</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Differential Expression Analysis of BRCA samples</h1>
<p class="author"><em>Cristina Sotomayor Vivas</em></p>
</div>
<div id="overview" class="section level1 unnumbered">
<h1>Overview</h1>
<p>This document contains the final project for the “Introduction to RNA-Seq” module, part of the Bioinformatics and Statistics II class given at LCG-UNAM in February 2022. The original class material from <a href="http://lcolladotor.github.io/">Dr. Leonardo Collado-Torres</a> can be found <a href="https://lcolladotor.github.io/rnaseq_LCG-UNAM_2022/index.html">here</a>.</p>
<p>In this project, we will use the <a href="http://rna.recount.bio"><code>recount3</code></a> R package to download RNA-Seq data from The Cancer Genome Atlas project, obtaining all the BRCA samples. We will then evaluate the quality of the downloaded data, explore the available sample attributes, select a subset of these to build a statistical model, and perform differential expression analysis.</p>
<div id="requirements" class="section level2 unnumbered">
<h2>Requirements</h2>
<p>The following libraries are used in this analysis.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="index.html#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;sessioninfo&quot;</span>)</span>
<span id="cb1-2"><a href="index.html#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="index.html#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co"># To download data and generate RangedSummarizedExperiment data object  </span></span>
<span id="cb1-4"><a href="index.html#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;recount3&quot;</span>)</span>
<span id="cb1-5"><a href="index.html#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="index.html#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co"># To access data on tumor subtype </span></span>
<span id="cb1-7"><a href="index.html#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;TCGAbiolinks&quot;</span>)</span>
<span id="cb1-8"><a href="index.html#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="index.html#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="co"># To normalize counts</span></span>
<span id="cb1-10"><a href="index.html#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;edgeR&quot;</span>) </span>
<span id="cb1-11"><a href="index.html#cb1-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-12"><a href="index.html#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co"># To plot results </span></span>
<span id="cb1-13"><a href="index.html#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb1-14"><a href="index.html#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;pheatmap&quot;</span>)</span>
<span id="cb1-15"><a href="index.html#cb1-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-16"><a href="index.html#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="co"># For differential expression analysis</span></span>
<span id="cb1-17"><a href="index.html#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;limma&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="index.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Reproducibility information</span></span>
<span id="cb2-2"><a href="index.html#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">width =</span> <span class="dv">120</span>)</span>
<span id="cb2-3"><a href="index.html#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.1.2 (2021-11-01)
##  os       macOS Monterey 12.1
##  system   x86_64, darwin17.0
##  ui       RStudio
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       America/Mexico_City
##  date     2022-02-10
##  rstudio  2021.09.2+382 Ghost Orchid (desktop)
##  pandoc   2.14.0.3 @ /Applications/RStudio.app/Contents/MacOS/pandoc/ (via rmarkdown)
## 
## ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
##  package              * version    date (UTC) lib source
##  AnnotationDbi          1.56.2     2021-11-09 [1] Bioconductor
##  assertthat             0.2.1      2019-03-21 [1] CRAN (R 4.1.0)
##  Biobase              * 2.54.0     2021-10-26 [1] Bioconductor
##  BiocFileCache          2.2.1      2022-01-23 [1] Bioconductor
##  BiocGenerics         * 0.40.0     2021-10-26 [1] Bioconductor
##  BiocIO                 1.4.0      2021-10-26 [1] Bioconductor
##  BiocParallel           1.28.3     2021-12-09 [1] Bioconductor
##  biomaRt                2.50.3     2022-02-03 [1] Bioconductor
##  Biostrings             2.62.0     2021-10-26 [1] Bioconductor
##  bit                    4.0.4      2020-08-04 [1] CRAN (R 4.1.0)
##  bit64                  4.0.5      2020-08-30 [1] CRAN (R 4.1.0)
##  bitops                 1.0-7      2021-04-24 [1] CRAN (R 4.1.0)
##  blob                   1.2.2      2021-07-23 [1] CRAN (R 4.1.0)
##  bookdown               0.24       2021-09-02 [1] CRAN (R 4.1.0)
##  bslib                  0.3.1      2021-10-06 [1] CRAN (R 4.1.0)
##  cachem                 1.0.6      2021-08-19 [1] CRAN (R 4.1.0)
##  cli                    3.1.1      2022-01-20 [1] CRAN (R 4.1.2)
##  colorspace             2.0-2      2021-06-24 [1] CRAN (R 4.1.0)
##  crayon                 1.4.2      2021-10-29 [1] CRAN (R 4.1.0)
##  curl                   4.3.2      2021-06-23 [1] CRAN (R 4.1.0)
##  data.table             1.14.2     2021-09-27 [1] CRAN (R 4.1.0)
##  DBI                    1.1.2      2021-12-20 [1] CRAN (R 4.1.0)
##  dbplyr                 2.1.1      2021-04-06 [1] CRAN (R 4.1.0)
##  DelayedArray           0.20.0     2021-10-26 [1] Bioconductor
##  digest                 0.6.29     2021-12-01 [1] CRAN (R 4.1.0)
##  downloader             0.4        2015-07-09 [1] CRAN (R 4.1.0)
##  dplyr                  1.0.7      2021-06-18 [1] CRAN (R 4.1.0)
##  edgeR                * 3.36.0     2021-10-26 [1] Bioconductor
##  ellipsis               0.3.2      2021-04-29 [1] CRAN (R 4.1.0)
##  evaluate               0.14       2019-05-28 [1] CRAN (R 4.1.0)
##  fansi                  1.0.2      2022-01-14 [1] CRAN (R 4.1.2)
##  farver                 2.1.0      2021-02-28 [1] CRAN (R 4.1.0)
##  fastmap                1.1.0      2021-01-25 [1] CRAN (R 4.1.0)
##  filelock               1.0.2      2018-10-05 [1] CRAN (R 4.1.0)
##  generics               0.1.2      2022-01-31 [1] CRAN (R 4.1.2)
##  GenomeInfoDb         * 1.30.1     2022-01-30 [1] Bioconductor
##  GenomeInfoDbData       1.2.7      2022-02-01 [1] Bioconductor
##  GenomicAlignments      1.30.0     2021-10-26 [1] Bioconductor
##  GenomicRanges        * 1.46.1     2021-11-18 [1] Bioconductor
##  ggplot2              * 3.3.5      2021-06-25 [1] CRAN (R 4.1.0)
##  glue                   1.6.1      2022-01-22 [1] CRAN (R 4.1.2)
##  gtable                 0.3.0      2019-03-25 [1] CRAN (R 4.1.0)
##  here                   1.0.1      2020-12-13 [1] CRAN (R 4.1.0)
##  highr                  0.9        2021-04-16 [1] CRAN (R 4.1.0)
##  hms                    1.1.1      2021-09-26 [1] CRAN (R 4.1.0)
##  htmltools              0.5.2      2021-08-25 [1] CRAN (R 4.1.0)
##  httr                   1.4.2      2020-07-20 [1] CRAN (R 4.1.0)
##  IRanges              * 2.28.0     2021-10-26 [1] Bioconductor
##  jquerylib              0.1.4      2021-04-26 [1] CRAN (R 4.1.0)
##  jsonlite               1.7.3      2022-01-17 [1] CRAN (R 4.1.2)
##  KEGGREST               1.34.0     2021-10-26 [1] Bioconductor
##  knitr                  1.37       2021-12-16 [1] CRAN (R 4.1.0)
##  labeling               0.4.2      2020-10-20 [1] CRAN (R 4.1.0)
##  lattice                0.20-45    2021-09-22 [1] CRAN (R 4.1.2)
##  lifecycle              1.0.1      2021-09-24 [1] CRAN (R 4.1.0)
##  limma                * 3.50.0     2021-10-26 [1] Bioconductor
##  locfit                 1.5-9.4    2020-03-25 [1] CRAN (R 4.1.0)
##  magrittr               2.0.2      2022-01-26 [1] CRAN (R 4.1.2)
##  Matrix                 1.4-0      2021-12-08 [1] CRAN (R 4.1.0)
##  MatrixGenerics       * 1.6.0      2021-10-26 [1] Bioconductor
##  matrixStats          * 0.61.0     2021-09-17 [1] CRAN (R 4.1.0)
##  memoise                2.0.1      2021-11-26 [1] CRAN (R 4.1.0)
##  munsell                0.5.0      2018-06-12 [1] CRAN (R 4.1.0)
##  pheatmap             * 1.0.12     2019-01-04 [1] CRAN (R 4.1.0)
##  pillar                 1.7.0      2022-02-01 [1] CRAN (R 4.1.2)
##  pkgconfig              2.0.3      2019-09-22 [1] CRAN (R 4.1.0)
##  plyr                   1.8.6      2020-03-03 [1] CRAN (R 4.1.0)
##  png                    0.1-7      2013-12-03 [1] CRAN (R 4.1.0)
##  prettyunits            1.1.1      2020-01-24 [1] CRAN (R 4.1.0)
##  progress               1.2.2      2019-05-16 [1] CRAN (R 4.1.0)
##  purrr                  0.3.4      2020-04-17 [1] CRAN (R 4.1.0)
##  R.methodsS3            1.8.1      2020-08-26 [1] CRAN (R 4.1.0)
##  R.oo                   1.24.0     2020-08-26 [1] CRAN (R 4.1.0)
##  R.utils                2.11.0     2021-09-26 [1] CRAN (R 4.1.0)
##  R6                     2.5.1      2021-08-19 [1] CRAN (R 4.1.0)
##  rappdirs               0.3.3      2021-01-31 [1] CRAN (R 4.1.0)
##  RColorBrewer           1.1-2      2014-12-07 [1] CRAN (R 4.1.0)
##  Rcpp                   1.0.8      2022-01-13 [1] CRAN (R 4.1.2)
##  RCurl                  1.98-1.5   2021-09-17 [1] CRAN (R 4.1.0)
##  readr                  2.1.2      2022-01-30 [1] CRAN (R 4.1.2)
##  recount3             * 1.4.0      2021-10-26 [1] Bioconductor
##  restfulr               0.0.13     2017-08-06 [1] CRAN (R 4.1.0)
##  rjson                  0.2.21     2022-01-09 [1] CRAN (R 4.1.2)
##  rlang                  1.0.1      2022-02-03 [1] CRAN (R 4.1.2)
##  rmarkdown              2.11       2021-09-14 [1] CRAN (R 4.1.0)
##  rprojroot              2.0.2      2020-11-15 [1] CRAN (R 4.1.0)
##  Rsamtools              2.10.0     2021-10-26 [1] Bioconductor
##  RSQLite                2.2.9      2021-12-06 [1] CRAN (R 4.1.0)
##  rstudioapi             0.13       2020-11-12 [1] CRAN (R 4.1.0)
##  rtracklayer            1.54.0     2021-10-26 [1] Bioconductor
##  rvest                  1.0.2      2021-10-16 [1] CRAN (R 4.1.0)
##  S4Vectors            * 0.32.3     2021-11-21 [1] Bioconductor
##  sass                   0.4.0.9000 2022-02-02 [1] Github (rstudio/sass@f7a9540)
##  scales                 1.1.1      2020-05-11 [1] CRAN (R 4.1.0)
##  sessioninfo          * 1.2.2      2021-12-06 [1] CRAN (R 4.1.0)
##  stringi                1.7.6      2021-11-29 [1] CRAN (R 4.1.0)
##  stringr                1.4.0      2019-02-10 [1] CRAN (R 4.1.0)
##  SummarizedExperiment * 1.24.0     2021-10-26 [1] Bioconductor
##  TCGAbiolinks         * 2.22.4     2022-01-23 [1] Bioconductor
##  TCGAbiolinksGUI.data   1.14.0     2021-10-30 [1] Bioconductor
##  tibble                 3.1.6      2021-11-07 [1] CRAN (R 4.1.0)
##  tidyr                  1.2.0      2022-02-01 [1] CRAN (R 4.1.2)
##  tidyselect             1.1.1      2021-04-30 [1] CRAN (R 4.1.0)
##  tzdb                   0.2.0      2021-10-27 [1] CRAN (R 4.1.0)
##  utf8                   1.2.2      2021-07-24 [1] CRAN (R 4.1.0)
##  vctrs                  0.3.8      2021-04-29 [1] CRAN (R 4.1.0)
##  withr                  2.4.3      2021-11-30 [1] CRAN (R 4.1.0)
##  xfun                   0.29       2021-12-14 [1] CRAN (R 4.1.0)
##  XML                    3.99-0.8   2021-09-17 [1] CRAN (R 4.1.0)
##  xml2                   1.3.3      2021-11-30 [1] CRAN (R 4.1.0)
##  XVector                0.34.0     2021-10-26 [1] Bioconductor
##  yaml                   2.2.2      2022-01-25 [1] CRAN (R 4.1.2)
##  zlibbioc               1.40.0     2021-10-26 [1] Bioconductor
## 
##  [1] /Library/Frameworks/R.framework/Versions/4.1/Resources/library
## 
## ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="index.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">proc.time</span>()</span></code></pre></div>
<pre><code>##      user    system   elapsed 
##   623.514    81.855 11607.629</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="downloading-data.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
